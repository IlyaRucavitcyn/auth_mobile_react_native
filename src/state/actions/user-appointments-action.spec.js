import
    * as userAppointmentsAction
    from "./user-appointments-action";
jest.mock('../../services/firebase-client', () => ({
    getClient: () => ({
        pushValue: () => Promise.resolve()
    })
}))


describe("user appointments action creator testing", () => {
    const appointments = ["test_1", "test_2"];
    describe("setUserAppointmentsAction method", () => {

        test("should return action with right type and payload", () => {
            const action = userAppointmentsAction
                .setUserAppointmentsAction(appointments);
            expect(action).toMatchSnapshot()
        });
    })

    describe("addUserAppointmentsAction method", () => {
        const uid = "testing_uid";

        test("dispatch should be called when firebase pushValue resolved", async () => {
            const resultingThump = userAppointmentsAction
                .addUserAppointmentsAction(uid, appointments);
            const dispatch = jest.fn();
            await resultingThump(dispatch);
            expect(dispatch).toHaveBeenCalled();
            const action = dispatch.mock.calls[0][0];
            expect(action).toMatchSnapshot();
        });

        test.skip("dispatch should be called when firebase pushValue rejected", async () => {
            jest.resetModules();
            jest.doMock('../../services/firebase-client', () => ({
                getClient: () => ({
                    pushValue: () => Promise.reject()

                })
            }))
            const resultingThump = userAppointmentsAction
                .addUserAppointmentsAction(uid, appointments);
            const dispatch = jest.fn();
            await resultingThump(dispatch);
            expect(dispatch).toHaveBeenCalledWith(undefined);
        });
    })

})