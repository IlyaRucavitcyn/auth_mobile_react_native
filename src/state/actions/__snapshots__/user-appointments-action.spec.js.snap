// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user appointments action creator testing addUserAppointmentsAction method dispatch should be called when firebase pushValue resolved 1`] = `
Object {
  "payload": Array [
    "test_1",
    "test_2",
  ],
  "type": "add-user-appointments",
}
`;

exports[`user appointments action creator testing setUserAppointmentsAction method should return action with right type and payload 1`] = `
Object {
  "payload": Array [
    "test_1",
    "test_2",
  ],
  "type": "set-user-appointments",
}
`;
